packages/taco-auth/src/providers/eip4361/eip4361.ts

```diff
@@ -75,37 +130,39 @@ export class EIP4361AuthProvider {
     return authMessage;
   }
 
-  private isMessageExpired(message: string): boolean {
-    const siweMessage = new SiweMessage(message);
-    if (!siweMessage.issuedAt) {
-      // don't expect to ever happen; but just in case
-      return false;
-    }
-
-    const twoHourWindow = new Date(siweMessage.issuedAt);
-    twoHourWindow.setHours(twoHourWindow.getHours() + 2);
-    const now = new Date();
-    return twoHourWindow < now;
-  }
-
+  /**
+   * Creates a new Sign-In with Ethereum (SIWE) authentication message and signature.
+   *
+   * Generates a new SIWE message using the configured parameters and current chain state.
+   * The message includes the user's address, current chainId, and other required SIWE fields.
+   * After creation, the message is signed by the user's wallet and verified before returning.
+   *
+   * @private
+   * @returns {Promise<EIP4361AuthSignature>} A newly created and signed authentication signature
+   * @throws {Error} If message signing fails or verification fails
+   */
   private async createSIWEAuthMessage(): Promise<EIP4361AuthSignature> {
     const address = await this.signer.getAddress();
-    const { domain, uri } = this.providerParams;
-    const version = '1';
-    const nonce = generateNonce();
     const chainId = (await this.provider.getNetwork()).chainId;
+
     const siweMessage = new SiweMessage({
-      domain,
+      statement: `${this.params.domain} wants you to sign in with your Ethereum account: ${address}`,
+      ...this.params,
       address,
-      statement: `${domain} wants you to sign in with your Ethereum account: ${address}`,
-      uri,
-      version,
-      nonce,
       chainId,
+      // if the user does not provide an expirationTime, we set it to 2 hours from now
+      expirationTime: new Date(
+        Date.now() + this.expirationDuration,
+      ).toISOString(),
     });
     const scheme = EIP4361_AUTH_METHOD;
+
     const message = siweMessage.prepareMessage();
     const signature = await this.signer.signMessage(message);
+
+    // this will trigger validation for all parameters including `expirationTime` and `notBefore`.
+    await siweMessage.verify({ signature });
```

Comment:
should we add a .catch() here? I mean, what happens if this verification fail? Should we return an error?